test/relations.test.js

```diff
@@ -445,6 +469,34 @@ describe('relations', function() {
               should.exist(ch);
               ch.id.should.eql(id);
               ch.name.should.equal('aa');
+              ch.type.should.equal('b');
+              done();
+            });
+        }
+      });
+
+      it('should replace scoped record with promises', function(done) {
+        var id;
+        Book.create()
```

Comment:
nitpick(Not a big deal feel free to ignore): To simplify the test code I think you can omit `done` here.
```js
it('should replace...', function() {
  return Book.create()
    .then(function(book) {
      // create chapters
    })
    .then(function(ch) {
       return verify(book);
     });

    function verify(book) {
      return book.chapters.findById(id)
        .then(function(ch) {
          // verify chapter
         });
    }
})
```